api/documents/document_ref.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package documents
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+	v1 "github.com/nitrictech/go-sdk/interfaces/nitric/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// DocumentRef - Represents a reference to a document
+type DocumentRef interface {
```

Comment:
Should probably have an Id() method to return the Document ID. Primarily for the refs included in get/query responses.

Reply:
Also a Parent() method or similar to retrieve the containing CollectionRef

Reply:
Done